### PR TITLE
Pass phishfort specific new issue url to phishing-warning page.

### PR DESCRIPTION
**Pull request by danjm**
Compares to https://github.com/MetaMask/metamask-extension/pull/14906. We will merge this into that before that is merged.

This PR is part of the resolution to this comment https://github.com/MetaMask/metamask-extension/pull/14906#issuecomment-1164472625 and https://github.com/MetaMask/phishing-warning/issues/15

Along with https://github.com/MetaMask/phishing-warning/pull/23, this PR ensures that users who land on the phishing warning page because they were attempting to access a url blocked by phishfort are redirected to the phishfort github issues page when clicking the "please file an issue" link

### CODE REVIEW

File: app/scripts/contentscript.js [hunk 1]
```diff
@@ -358,12 +358,15 @@ function blockedDomainCheck() {
...
+ * @param data
  */
-function redirectToPhishingWarning() {
+function redirectToPhishingWarning(data) {
   console.debug('MetaMask: Routing to Phishing Warning page.');
   const baseUrl = process.env.PHISHING_WARNING_PAGE_URL;
   window.location.href = `${baseUrl}#${querystring.stringify({
     hostname: window.location.hostname,
     href: window.location.href,
+    newIssueUrl: data.newIssueUrl,
```
Comment (digiwand) [inline app/scripts/contentscript.js:370]:
```suggestion
    newIssueUrl: data?.newIssueUrl,
```

**Review by digiwand — APPROVED** [review]
One suggestion and I think the rest looks good

**Comment by darkwing** [comment]
Test failure appears to be legit

File: test/e2e/tests/phishing-detection.spec.js [hunk 2]
```diff
@@ -127,4 +138,30 @@ describe('Phishing Detection', function () {
...
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+        await driver.openNewPage('http://127.0.0.1:8080');
+        const newIssueLink = await driver.findElement({
+          text: 'please file an issue',
+        });
+        assert.equal(
+          await newIssueLink.getAttribute('href'),
```
Comment (NiranjanaBinoy) [inline test/e2e/tests/phishing-detection.spec.js:161]: The `href` value here might be the new GitHub ticket link that will be created to report the issue.

**Review by NiranjanaBinoy — APPROVED** [review]

**Pull request merged**